Claude Agent SDK (#142)
airbytehq/quickstarts: agents_examples/claude_agent_sdk/README.md

# Bug-triage agent: Claude Agent SDK + Airbyte Agents

A runnable example agent that reads new GitHub issues, cross-references Linear for duplicates, and posts a triage summary to Slack. It shows how to give a [Claude Agent SDK](https://github.com/anthropics/claude-agent-sdk-python) agent live read and write access to your business systems through [Airbyte Agents](https://airbyte.com).

The agent lives in `airbyte-claude-agent-sdk-example.py`. You can wire each connector's Airbyte tools into the agent yourself, which gives you full programmatic control.

## Prerequisites

> - Python 3.10 or newer. The Claude Agent SDK bundles the Claude Code CLI, so there is nothing else to install for it to run.

Author: ```suggestion
- Python 3.11 or newer. The Claude Agent SDK bundles the Claude Code CLI, so there is nothing else to install for it to run.
```

## Run

```bash
uv run airbyte-claude-agent-sdk-example.py
```

The script prints a health line per connector, then streams the agent's work as it reads GitHub and Linear, then posts to Slack.

## How it works

Every Airbyte connector exposes three callables through `build_connector_tools(connector)`: `inspect_connector`, `read_skill_docs`, and `execute`. The agent uses them in a progressive flow:

1. `inspect_connector()` reports the connector's metadata and Context Store readiness, and resolves the skill-doc ID the other two calls use.
2. `read_skill_docs()` returns an outline of the connector's entities and actions. `read_skill_docs(section="...")` drills into the guidance for the specific operation the agent is about to run.
3. `execute(entity, action, params)` runs the operation and returns a structured result with `data` (the records) and `meta` (pagination cursors).

> [!NOTE]
> Airbyte Agents SDK offers a function called `build_connector_tools` to encapsulate all three calls for some frameworks. However, The Claude Agent SDK is not one of the frameworks, so the `make_airbyte_tools` helper takes each callable and wraps it as a custom `@tool`. 

> Tool names are prefixed per connector (for example `github_execute`) because the three callables share a name across connectors, and all of them register on a single in-process MCP server. `allowed_tools=["mcp__airbyte__*"` pre-approves every tool on that server so none trip a permission prompt.

Author: ```suggestion
Tool names are prefixed per connector (for example `github_execute`) because the three callables share a name across connectors, and all of them register on a single in-process MCP server. `allowed_tools=["mcp__airbyte__*"]` pre-approves every tool on that server so none trip a permission prompt.
```

Handlers catch exceptions and return `is_error: True`. When the agent guesses a wrong `read_skill_docs` section, the error message carries the valid outline for Claude to read and retry without needing to abort.

> 

Author: ````suggestion

## Add connectors

In `main`, add a tuple to the `connectors` list with the following syntax:

```python
        ("<connector_name>", connect("<connector_name>")),
```
````

2. `read_skill_docs()` returns an outline of the connector's entities and actions. `read_skill_docs(section="...")` drills into the guidance for the specific operation the agent is about to run.
3. `execute(entity, action, params)` runs the operation and returns a structured result with `data` (the records) and `meta` (pagination cursors).

> [!NOTE]
> Airbyte Agents SDK offers a function called `build_connector_tools` to encapsulate all three calls for some frameworks. However, The Claude Agent SDK is not one of the frameworks, so the `make_airbyte_tools` helper takes each callable and wraps it as a custom `@tool`. 

Tool names are prefixed per connector (for example `github_execute`) because the three callables share a name across connectors, and all of them register on a single in-process MCP server. `allowed_tools=["mcp__airbyte__*"]` pre-approves every tool on that server so none trip a permission prompt.

Handlers catch exceptions and return `is_error: True`. When the agent guesses a wrong `read_skill_docs` section, the error message carries the valid outline for Claude to read and retry without needing to abort.

## Add connectors

> In `main`, add a tuple to the `connectors` list with the following syntax:

Author: ````suggestion
In `main`, add the name of the connector to the `connectors` list by adding it to the tuple in line `124`:

```python
    for slug in ("github", "linear", "slack"):
```
````

# Bug-triage agent: Claude Agent SDK + Airbyte Agents

A runnable example agent that reads new GitHub issues, cross-references Linear for duplicates, and posts a triage summary to Slack. It shows how to give a [Claude Agent SDK](https://github.com/anthropics/claude-agent-sdk-python) agent live read and write access to your business systems through [Airbyte Agents](https://airbyte.com).

The agent lives in `airbyte-claude-agent-sdk-example.py`. You can wire each connector's Airbyte tools into the agent yourself, which gives you full programmatic control.

## Prerequisites

- Python 3.11 or newer. The Claude Agent SDK bundles the Claude Code CLI, so there is nothing else to install for it to run.
- An Airbyte account with the GitHub, Linear, and Slack connectors connected in your workspace. The Free tier is enough to follow along. Airbyte Agents ships 50+ agent connectors, so you can swap in others; confirm what is live with `list_connectors()`.
- An Anthropic API key and your Airbyte workspace credentials (see below).
- `uv` for dependency management. `pip` works too.

## Setup

Install the two SDKs and `python-dotenv`:

> ```bash

Author: ````suggestion
```bash
uv init claude_agent_sdk
````

3. `execute(entity, action, params)` runs the operation and returns a structured result with `data` (the records) and `meta` (pagination cursors).

> [!NOTE]
> Airbyte Agents SDK offers a function called `build_connector_tools` to encapsulate all three calls for some frameworks. However, The Claude Agent SDK is not one of the frameworks, so the `make_airbyte_tools` helper takes each callable and wraps it as a custom `@tool`. 

Tool names are prefixed per connector (for example `github_execute`) because the three callables share a name across connectors, and all of them register on a single in-process MCP server. `allowed_tools=["mcp__airbyte__*"]` pre-approves every tool on that server so none trip a permission prompt.

Handlers catch exceptions and return `is_error: True`. When the agent guesses a wrong `read_skill_docs` section, the error message carries the valid outline for Claude to read and retry without needing to abort.

## Add connectors

In `main`, add a tuple to the `connectors` list with the following syntax:

> 

Author: ````suggestion

```python
("<connector>", connect("<connector>")),
```
````
